complex/src/scalar.rs

```diff
@@ -0,0 +1,252 @@
+
+use traits::{Num, FromPrimitive, Float};
+use std::ops::Neg;
+
+use Complex;
+
+pub trait Scalar: Num + Copy + Neg<Output = Self> {
+    /// Associated Real type
+    type Real;
+    /// Associated Complex type
+    type Complex;
+
+    /// Take the square root of a number.
+    fn sqrt(&self) -> Self;
+    /// Returns `e^(self)`, (the exponential function).
+    fn exp(&self) -> Self;
```

Comment:
nit: please put a blank line between all `type`/`fn` items.  If you want visual clustering of related items, you can use two blank lines around them.